editor/sidebar/post-taxonomies/index.js

```diff
@@ -24,58 +24,37 @@ import { toggleSidebarPanel } from '../../actions';
  */
 const PANEL_NAME = 'post-taxonomies';
 
-class PostTaxonomies extends Component {
-	constructor() {
-		super( ...arguments );
+function PostTaxonomies( { postType, taxonomies, isOpened, onTogglePanel } ) {
+	const availableTaxonomies = !! taxonomies.data
+		? Object.values( taxonomies.data ).filter( ( taxonomy ) => taxonomy.types.indexOf( postType ) !== -1 )
+		: [];
 
-		this.state = {
-			taxonomies: [],
-		};
-	}
-
-	componentDidMount() {
-		this.fetchTaxonomies = new wp.api.collections.Taxonomies()
-			.fetch()
-			.done( ( taxonomies ) => {
-				this.setState( { taxonomies: Object.values( taxonomies ) } );
-			} );
-	}
-
-	componentWillUnmout() {
-		this.fetchTaxonomies.abort();
+	if ( ! availableTaxonomies.length ) {
+		return null;
 	}
 
-	render() {
-		const availableTaxonomies = this.state.taxonomies
-			.filter( ( taxonomy ) => taxonomy.types.indexOf( this.props.postType ) !== -1 );
-
-		if ( ! availableTaxonomies.length ) {
-			return null;
-		}
-
-		return (
-			<PanelBody
-				title={ __( 'Categories & Tags' ) }
-				opened={ this.props.isOpened }
-				onToggle={ this.props.onTogglePanel }
-			>
-				{ availableTaxonomies.map( ( taxonomy ) => {
-					const TaxonomyComponent = taxonomy.hierarchical ? HierarchicalTermSelector : FlatTermSelector;
-					return (
-						<TaxonomyComponent
-							key={ taxonomy.slug }
-							label={ taxonomy.name }
-							restBase={ taxonomy.rest_base }
-							slug={ taxonomy.slug }
-						/>
-					);
-				} ) }
-			</PanelBody>
-		);
-	}
+	return (
+		<PanelBody
+			title={ __( 'Categories & Tags' ) }
+			opened={ isOpened }
+			onToggle={ onTogglePanel }
+		>
+			{ availableTaxonomies.map( ( taxonomy ) => {
```

Comment:
To the above point, Lodash's `reduce` similarly handles `undefined` and `object` types, so this could even manage the filtering and mapping in one, ensuring the panel heading is always shown.

Reply:
you know I prefer `filter` + `map` over `reduce` on small loops. I do understand the performance reasons for long loops.

Reply:
I meant nothing of performance by my comment, `filter` / `map` or whatever is fine, just that we can skip the early return and account for `undefined` type with Lodash methods.